### PR TITLE
add Popper & Reference innerRef prop

### LINKED ISSUE #140: Getting ref to Popper element in v1.0.0-beta.5
### Reproduction demo
https://codesandbox.io/s/10nmjoq5jq

### What is the expected behavior?
Is it possible with the current v1.0.0 api to get a reference to the popper element? In v0.10.1, an innerRef function existed that could be passed into the Popper component, and would be forwarded onto the ref passed down to children: https://github.com/FezVrasta/react-popper/blob/v0.10.1/src/Popper.js#L188

### What went wrong?
On v1.0.0-beta.5, the innerRef prop is removed and trying to set it manually results in an infinite loop, since the ref callback is putting the ref in state: https://github.com/FezVrasta/react-popper/blob/master/src/Popper.js#L71, triggering componentDidUpdate.

### Packages versions
- react-popper: v1.0.0-beta.5

### PR DESCRIPTION
**Pull request by giladgray**
#### Fixes #140 

tested this in Blueprint's `Popover` locally (via `yarn link`) and was able to get it rendering _without_ the infinite loop!

@FezVrasta this was my first experience with Flow and I must say it was rather subpar compared to TypeScript. Why did you choose to use Flow for this project and would you consider migrating to TypeScript? (I'd be happy to do the conversion, it'll take like an hour.)

### CODE REVIEW

**Comment by FezVrasta** [comment]
Thanks for the PR, the decision to use Flow has been carefully pondered and I'm not willing to switch to TypeScript.

Some of the reasons are:
- TS is not JS
- TS is not a sound type system 
- Flow inference is way better than TS

File: src/utils.js [hunk 1]
```diff
@@ -1,5 +1,17 @@
 // @flow
 
-// Takes an argument and if it's an array, returns the first item in the array,
-// otherwise returns the argument. Used for Preact compatibility.
+/**
```
[the remark this answers is not in the document]
Comment (giladgray) [reply src/utils.js]: are you sure? worth noting that `/**` comments appear in editor tooltips so you can see the usage docs when you actually use the function.

**Comment by giladgray** [comment]
🤔 I would argue that Flow is not JS either, but the `.js` extension confusingly suggests that it is: you can't actually execute the code until you've run it through a transpiler and removed the non-JS `type` statements! The `.ts` extension makes this extra step obvious, plus the TS compiler is basically Babel + Flow + polyfills in one tool. TS is definitely JS (with purely additive type information) -- it deliberately adheres to the latest ECMAScript specs and provides consistent support for upcoming actual JS features long before they're widely available in browsers.

**Comment by FezVrasta** [comment]
You can write Flow annotations as JS comments. The point is that once the types are stipped, the resulting code is javascript.

TypeScript on the other hand is a superset of JavaScript, it introduces new language features such as the public and private class properties. Once comments are stripped, you don't get valid JavaScript

**Comment by TrySound** [comment]
TS is also hard to swallow for the first time. With experience you see more advantages.

File: src/utils.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (FezVrasta) [reply src/utils.js]: What editor are you talking about? in VSCode I don't see any difference.

[the remark this answers is not in the document]
Comment (giladgray) [reply src/utils.js]: yes VS Code. you see these `//` comments in the tooltip when you hover over a usage? that may be a Flow thing... feels pretty non-standard.

[the remark this answers is not in the document]
Comment (FezVrasta) [reply src/utils.js]: I don't see the hints with neither of them, anyway it's okay to use this other format, but we should be consistent in the whole code-base

**Comment by giladgray** [comment]
@FezVrasta flow discussion aside, any more work required on this PR?

**Comment by giladgray** [comment]
@FezVrasta ping on this? would love to ship this and start using it in Blueprint.

**Comment by FezVrasta** [comment]
I'm sorry, been a bit busy lately

**Comment by giladgray** [comment]
thanks @FezVrasta!! i suppose the next step is to ship beta.7?

**Pull request merged**